pySDC/helpers/fieldsIO.py

```diff
@@ -515,6 +517,21 @@ def setupMPI(cls, comm: MPI.Intracomm, iLoc, nLoc):
         cls.nLoc = nLoc
         cls.mpiFile = None
 
+    @property
```

Comment:
Probably better to call the allreduce in `setupMPI` and store `numCollectiveIO` in one attribute at this point

Reply:
Well, if you want that, then you need to pass the instance or the number of variables in `setupMPI`. I went this way in order not to change the interface.

Reply:
Oh yes, I forgot that ... then put `cls._num_collective_IO` to None in `setupMPI`, to avoid any bug if two different decomposition are used one after the other